### PR TITLE
update GMC e2e and Doc

### PR DESCRIPTION
**Pull request by KfreeZ**
## Description

1. fix minor errors
3. sync to Gaudi scripts
4. add doc 

## Issues

 `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

 `n/a`.

## Tests

 CI/CD will cover the tests

### CODE REVIEW

File: .github/workflows/scripts/e2e/gmc_gaudi_test.sh [hunk 1]
```diff
@@ -39,9 +46,58 @@ function validate_gmc() {
...
+    echo "validate webhook"
+    validate_webhook
+
     get_gmc_controller_logs
 }
 
+function validate_webhook() {
+    kubectl create ns $WEBHOOK_NAMESPACE || echo "namespace $WEBHOOK_NAMESPACE is created."
+    # validate root node existence
+    yq ".spec.nodes.node123 = .spec.nodes.root | del(.spec.nodes.root)" config/samples/chatQnA_xeon.yaml > /tmp/webhook-case1.yaml
```
Comment (ichbinblau) [inline .github/workflows/scripts/e2e/gmc_gaudi_test.sh:58]: Shall we validate based on gaudi sample yamls?

File: .github/workflows/scripts/e2e/gmc_gaudi_test.sh [hunk 2]
```diff
@@ -39,9 +46,58 @@ function validate_gmc() {
...
+        exit 1
+    fi
+
+    # clean up cases
+    rm -f /tmp/webhook-case*.yaml
+}
+
 function cleanup_apps() {
     echo "clean up microservice-connector"
     # namespaces=("$CHATQNA_NAMESPACE" "$CHATQNA_DATAPREP_NAMESPACE" "$CHATQNA_SWITCH_NAMESPACE" "$CODEGEN_NAMESPACE" "$CODETRANS_NAMESPACE" "$DOCSUM_NAMESPACE")
```
Comment (ichbinblau) [inline .github/workflows/scripts/e2e/gmc_gaudi_test.sh:103]: Plz remove the WEBHOOK_NAMESPACE in func cleanup_apps.

Comment (KfreeZ) [reply .github/workflows/scripts/e2e/gmc_gaudi_test.sh:103]: done

**Review by ichbinblau — APPROVED** [review]
LGTM

**Comment by irisdingbj** [comment]
some nit, otherwise LGTM. Thank you!

**Review by irisdingbj — APPROVED** [review]

**Review by daisy-ycguo — APPROVED** [review]

**Pull request merged**